packages/ethers/src/client.ts

```diff
@@ -1255,19 +1255,13 @@ export class Web3Modal extends Web3ModalScaffold {
             EthersStoreUtil.setChainId(chainId)
             // eslint-disable-next-line @typescript-eslint/no-explicit-any
           } catch (switchError: any) {
-            if (
-              switchError.code === EthersConstantsUtil.ERROR_CODE_UNRECOGNIZED_CHAIN_ID ||
-              switchError.code === EthersConstantsUtil.ERROR_CODE_DEFAULT ||
-              switchError?.data?.originalError?.code ===
-                EthersConstantsUtil.ERROR_CODE_UNRECOGNIZED_CHAIN_ID
-            ) {
-              await EthersHelpersUtil.addEthereumChain(
-                WalletConnectProvider as unknown as Provider,
-                chain
-              )
-            } else {
+            if (/(?<temp1>user rejected)/u.test(switchError?.message)) {
```

Comment:
@ganchoradkov Why this has difference with the wagmi's own implementation? As I see the wagmi's code have case-insensivity while this one doesn't have:

```js
const switchError1 = { message: "user rejected" };
const switchError2 = { message: "User rejected" };

// First regex (case-sensitive, named group)
const regex1 = /(?<temp1>user rejected)/u;

// Second regex (case-insensitive)
const regex2 = /(user rejected)/i;

// Test cases
console.log(regex1.test(switchError1.message)); // true
console.log(regex2.test(switchError1.message)); // true

console.log(regex1.test(switchError2.message)); // false
console.log(regex2.test(switchError2.message)); // true
```

Do you think we might need this?

Reply:
appkit has extra ts rules that force `case-sensitive, named group`

Reply:
or we can go with `//ts-ignore` usage, anyway it is sifficient for the use-case they are updated

Reply:
fixed by lower casing the error msg before the check